### PR TITLE
Node/CCQ: Track concurrent queries

### PR DESCRIPTION
**Pull request by bruce-riley**
This PR adds a metric to the proxy server to track the maximum number of concurrent requests by chain. This will be useful for scaling the amount of parallelism in the guardian watchers.

### CODE REVIEW

File: node/cmd/ccq/metrics.go [hunk 1]
```diff
@@ -108,4 +111,19 @@ var (
 			Name: "ccq_server_total_number_of_successful_reconnects",
 			Help: "Total number of successful reconnects to bootstrap peers",
 		})
+
+	maxConcurrentQueriesByChain = promauto.NewGaugeVec(
```
Comment (evan-gray) [inline node/cmd/ccq/metrics.go:115]: I wonder if it's also worth tracking `currentConcurrentQueriesByChain` to see how bursty the overlapping requests are.

**Review by evan-gray — APPROVED** [review]

File: node/cmd/ccq/http.go [hunk 2]
```diff
@@ -112,7 +112,8 @@ func (s *httpServer) handleQuery(w http.ResponseWriter, r *http.Request) {
 		Signature:    signature,
 	}
 
-	if status, err := validateRequest(s.logger, s.env, s.permissions, s.signerKey, apiKey, signedQueryRequest); err != nil {
+	status, queryReq, err := validateRequest(s.logger, s.env, s.permissions, s.signerKey, apiKey, signedQueryRequest)
+	if err != nil {
```
Comment (panoel) [inline node/cmd/ccq/http.go:116]: Is it necessary to check if queryReq is nil?

**Review by panoel — APPROVED** [review]

File: node/cmd/ccq/http.go (hunk 2 above)
Comment (bruce-riley) [reply node/cmd/ccq/http.go:116]: The function always returns queryReq when the error is nil, so I don't think it's necessary to check the pointer for nil.

File: node/cmd/ccq/utils.go [hunk 3]
```diff
@@ -89,14 +89,14 @@ func validateRequest(logger *zap.Logger, env common.Environment, perms *Permissi
 	if err != nil {
```
Comment (bingyuyap) [inline node/cmd/ccq/utils.go:89]: does `queryRequest.Unmarshal()` return error if `len(qr.QueryRequest)` is 0? If not would that unmarshal it into an empty object?

Comment (bingyuyap) [reply node/cmd/ccq/utils.go:89]: it seems like `UnmarshalFromReader` does that. All good!

**Review by bingyuyap — APPROVED** [review]
LGTM!

File: node/cmd/ccq/utils.go (hunk 3 above)
Comment (bruce-riley) [reply node/cmd/ccq/utils.go:89]: Unmarshal from an empty buffer returns this error:
`failed to read message version: EOF`

**Pull request merged**